docs/function-format.md

```diff
@@ -58,42 +62,68 @@ The header keys and values would be populated with information about the functio
 
 Pros:
 
-* Streamable
+* Supports streaming
 * Common format
 
 Cons:
 
 * Requires a parsing library or fair amount of code to parse headers properly
 * Double parsing - headers + body (if body is to be parsed, such as json)
 
-#### JSON I/O Format (not implemented)
+#### JSON Input Format
 
 `--format json`
 
-The idea here is to keep the HTTP base structure, but make it a bit easier to parse by making the `request line` and `headers` a JSON struct.
-Eg:
+Fn accepts request data of the following format:
 
+```json
+{
+  "body": {
+    "some": "input"
+  }
+}
 ```
+
+Internally function receives data in following format:
+
+```json
 {
-  "request_url":"http://....",
-  "params": {
-    "blog_name": "yeezy"
+  "body": "{\"some\": \"input\"}",
```

Comment:
this is incorrect, provided your example body it will be:

```
"body":"{\n\t\"body\": {\n\t\t\"some\": \"input\"\n\t}\n}\n"
```

we could probably simplify a little, here ya go https://play.golang.org/p/srantfAg5f ;)